experimental/LieAlgebras/src/RootSystem.jl

```diff
@@ -0,0 +1,489 @@
+###############################################################################
+#
+#   Root Systems and Weights
+#
+###############################################################################
+
+mutable struct RootSystem
+  cartan_matrix::ZZMatrix # (generalized) Cartan matrix
+  #fw::QQMatrix # fundamental weights as linear combination of simple roots
+  positive_roots::Any #::Vector{RootSpaceElem} (cyclic reference)
+  weyl_group::Any     #::WeylGroup (cyclic reference)
```

Comment:
Couldn't this at least be, say, 
```suggestion
  positive_roots::Vector #::Vector{RootSpaceElem} (cyclic reference)
  weyl_group::CoxeterGroup     #::WeylGroup (cyclic reference)
```
The entries could be made concrete via type parameters. If it ever matters (right now I guess it doesn't).